cdap-securestore-spi/src/main/java/co/cask/cdap/securestore/spi/SecretsMetadataStore.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package co.cask.cdap.securestore.spi;
+
+import java.util.Map;
+
+/**
+ * Stores and retrieves metadata for the secrets.
+ *
+ * @param <T> the type of the metadata to be stored.
+ */
+public interface SecretsMetadataStore<T> {
+
+  /**
+   * Retrieves stored metadata for the secret.
+   *
+   * @param namespace the namespace to which secret belongs to
+   * @param name the name of the secret to be retrieved
+   * @return stored metadata
+   * @throws SecretNotFoundException if secret is not found in provided namespace
+   * @throws Exception if unable to retrieve the secret
+   */
+  T get(String namespace, String name) throws Exception;
+
+  /**
+   * Provides list of all the secrets for which metadata is stored.
+   *
+   * @param namespace the namespace to which secrets belong to
+   * @return map of secret to description. If there is no secret metadata stored in provided namespace,
```

Comment:
Is description a typo? 
the store method does not take a description. Will this be fetched from SecretsManager?

Reply:
No thats what SecureStore api returns for list.